envoy/http/header_validator.h

```diff
@@ -18,12 +21,39 @@ class HeaderValidator {
 public:
   virtual ~HeaderValidator() = default;
 
-  enum class HeaderEntryValidationResult { Accept, Reject };
+  // A class that holds either success condition or an error condition with tuple of
+  // action and error details.
+  template <typename ActionType> class Result {
+  public:
+    using Action = ActionType;
+
+    // Helper for constructing successful results
+    static Result success() { return Result(ActionType::Accept, absl::string_view()); }
```

Comment:
This needs a TODO for removal since it encourages empty string views, no?

Reply:
We only need details if header map was invalid, i.e. when result was either `reject` or `redirect`. If header map passed, the details are empty, which is what this method is used for.

Reply:
ugh sorry yes, my bad.